spirv_msl.cpp

```diff
@@ -9669,7 +9872,15 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 
 		break;
 	}
-
+	case OpSetMeshOutputsEXT:
+	{
+		flush_variable_declaration(builtin_mesh_primitive_indices_id);
+		statement("if (gl_LocalInvocationIndex == 0)");
+		begin_scope();
+		statement("spv_primitive_count = ", to_unpacked_expression(ops[1]), ";");
```

Comment:
There is no need for the primitive count to be threadgroup when shaders write to it. There can be a threadgroup variable in the wrapped main that is written before the barrier. It can just be plain thread.

Also, it's missing the vertex count. That is useful when copying out vertex data. No need to copy out unused vertices ... This fake builtin can just be a uvec2 really.

Reply:
In [spec](https://github.com/KhronosGroup/Vulkan-Docs/blob/main/proposals/VK_EXT_mesh_shader.adoc):
> The arguments are taken from the first invocation in each workgroup. Any invocation must execute this instruction no more than once and under uniform control flow. There must not be any control flow path to an output write that is not preceded by this instruction

My read on this: while application has to ensure uniform control flow, it doesn't have to provide same arguments from every invocation, as only first invocation matters.